editor/editor_file_system.cpp

```diff
@@ -1138,6 +1138,15 @@ void EditorFileSystem::_process_file_system(const ScannedDirectory *p_scan_dir,
 					}
 				}
 			}
+
+			if (fi->uid == ResourceUID::INVALID_ID && !ResourceLoader::has_custom_uid_support(path) && ResourceLoader::exists(path) && !FileAccess::exists(path + ".uid")) {
```

Comment:
```suggestion
			if (fi->uid == ResourceUID::INVALID_ID && ResourceLoader::exists(path) && !ResourceLoader::has_custom_uid_support(path) && !FileAccess::exists(path + ".uid")) {
```
`has_custom_uid_support()` also checks for existence, so in this order you can avoid extra check for non-Resource files.